### PR TITLE
Not for merge - Preview CMIP6 TAS Kerchunk

### PR DESCRIPTION
**Pull request by abarciauskas-bgse**

### CODE REVIEW

**Comment by abarciauskas-bgse** [comment]
@j08lue @anayeaye @slesaad @brianmfreitag this is a preview of the CMIP6 visualization via Kerchunk/NetCDF for feedback.

Questions:
* I've only included the historical data. There are 4 SSPs (ssp126, ssp245, ssp370, ssp585) available for 2015-2100. Do we want to include one or more of those? If we want to include multiple SSPs we can easily publish separate collections. It will be slightly more complex if we want to publish multiple zarr assets within the same STAC collection. The simplest thing to do for now is determine if we want to include SSP data at all and if so, which SSP.
* Could you review text (dataset description and layer description) content or should we have someone from the CMIP6 NEX GDDP approve the content?

NOTEs:
* I'm trying to determine if we can mask out the no data values (i.e. water bodies), but as per usual the data encoding is giving some trouble
* I'm using near-surface air temperature and I've used the `jet` colormap with scale from 200,325. The data encoding issue below is blocking me from determining the absolute min,max of the data but this looks ok for now.

Let me know if you have other thoughts, but once we're happy with the dataset (STAC and MDX) configuration, I think the next step would be to publish to the staging STAC and add to veda-config.

**Comment by anayeaye** [comment]
@abarciauskas-bgse this preview is awesome!


RE
> I've only included the historical data. There are 4 SSPs (ssp126, ssp245, ssp370, ssp585) available for 2015-2100. Do we want to include one or more of those? If we want to include multiple SSPs we can easily publish separate collections. It will be slightly more complex if we want to publish multiple zarr assets within the same STAC collection. The simplest thing to do for now is determine if we want to include SSP data at all and if so, which SSP.

Can we pilot the simplest case first (no SSPs)? It would be great if defer the SSP decisions another week so that we can discuss organization strategies for the incoming series of nex-gddp data which include many models and SSPs?

**Comment by abarciauskas-bgse** [comment]
Thanks @anayeaye. I have updated the preview with a new colormap and I think we are good to go since, thanks to @vincentsarago the nodata values have been masked. I think the next steps would be to:

1. Publish the CMIP6 dataset to the staging STAC
2. Add the notebook for how to generate CMIP6 kerchunk reference to veda-data
3. Add the notebook for how to generate CMIP6 STAC record to veda-docs
4. Add the data MDX file to veda-ui

@anayeaye @j08lue @slesaad do you agree we should put this demo in the dashboard?

**Comment by j08lue** [comment]
I'm all for adding this dataset.

Let's flesh out the details, though, please:

- [ ] Add more info about the dataset to the Dataset page - no need to have loads of flow text, but some explanation of the origin of this dataset, maybe some scientific references. Pls see the [GHG Center dataset pages](https://ghg-demo.netlify.app/data-catalog) for content/formatting ideas.
   
   image

- [ ] Add units (I guess it's `K`)

   image

Can you provide this information, please, @abarciauskas-bgse?

File: mock/datasets/cmip6-tas.data.mdx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  * Temporal Resolution: daily
+
+Source: [https://www.nccs.nasa.gov/services/data-collections/land-based-products/nex-gddp-cmip6](https://www.nccs.nasa.gov/services/data-collections/land-based-products/nex-gddp-cmip6)
+
+## AWS Public Dataset
+
+There are 2 datasets listed on the AWS Registry of Open Data for [NEX-GDDP-CMIP6](https://registry.opendata.aws/nex-gddp-cmip6/). First, there is an archive of NetCDF files from about 35 different climate models, each supplying historical and predicted values for up to 9 environment variables, daily, from 1950 to 2100. Second, there is an archive of COGs generated from the corresponding NetCDFs to support visualization via dynamic tiling using COGs. COGs are only available for 2 models. The COG archive also includes monthly aggregatations across those models.
+
+## About this dataset
+This dataset represents a subset of the NEX-GDDP-CMIP6 data. This dataset was constructed to demonstrate a method for visualizing the data without creating any data copies just for visualization.
```
Comment (j08lue) [inline mock/datasets/cmip6-tas.data.mdx:71]:
```suggestion
VEDA hosts a subset of the NEX-GDDP-CMIP6 data, primarily for demo purposes. With this dataset, we demonstrate a method for visualizing the data without creating any data copies just for visualization.
```

File: mock/datasets/cmip6-tas.data.mdx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+This demonstration of uses the [kerchunk](https://fsspec.github.io/kerchunk/) python library to create a file - often called a reference file - which is an index of the chunks of the data stored in the original [NetCDF](https://www.unidata.ucar.edu/software/netcdf/) files. This reference file is used by a dynamic tiling library ([titiler-xarray](https://github.com/developmentseed/titiler-xarray)) to create image tiles from the underlying data. The indexes were generated for the near-surface air temperature variable (TAS) for years 1950-2014, the historical extent of the NEX-GDDP-CMIP6 project.
```
Comment (j08lue) [inline mock/datasets/cmip6-tas.data.mdx:73]:
```suggestion
To prepare this dataset, we used the [kerchunk](https://fsspec.github.io/kerchunk/) python library to create a file - often called a reference file - which is an index of the chunks of the data stored in the original [NetCDF](https://www.unidata.ucar.edu/software/netcdf/) files. This reference file is used by a dynamic tiling library ([titiler-xarray](https://github.com/developmentseed/titiler-xarray)) to create image tiles from the underlying data. The indexes were generated for the near-surface air temperature variable (TAS) for years 1950-2014, the historical extent of the NEX-GDDP-CMIP6 project.
```

File: mock/datasets/cmip6-tas.data.mdx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      - '#f2cbb7'
+      - '#ee8468'
+      - '#b40426'
+---
+
+<Block>
+<Prose>
+# NASA Earth Exchange Global Daily Downscaled Projections (NEX-GDDP-CMIP6)
+
+The NEX-GDDP-CMIP6 dataset is comprised of global downscaled climate scenarios derived from the General Circulation Model (GCM) runs conducted under the Coupled Model Intercomparison Project Phase 6 (CMIP6) and across all four “Tier 1” greenhouse gas emissions scenarios known as Shared Socioeconomic Pathways (SSPs). The CMIP6 GCM runs were developed in support of the Sixth Assessment Report of the Intergovernmental Panel on Climate Change (IPCC AR6). This dataset includes downscaled projections from ScenarioMIP model runs for which daily scenarios were produced and distributed through the Earth System Grid Federation. The purpose of this dataset is to provide a set of global, high resolution, bias-corrected climate change projections that can be used to evaluate climate change impacts on processes that are sensitive to finer-scale climate gradients and the effects of local topography on climate conditions.
```
Comment (j08lue) [inline mock/datasets/cmip6-tas.data.mdx:51]:
```suggestion
The NEX-GDDP-CMIP6 dataset provide a set of global, high resolution, bias-corrected climate change projections that can be used to evaluate climate change impacts on processes that are sensitive to finer-scale climate gradients and the effects of local topography on climate conditions.

NEX-GDDP-CMIP6 is comprised of global downscaled climate scenarios derived from the General Circulation Model (GCM) runs conducted under the Coupled Model Intercomparison Project Phase 6 (CMIP6) and across all four “Tier 1” greenhouse gas emissions scenarios known as Shared Socioeconomic Pathways (SSPs). The CMIP6 GCM runs were developed in support of the Sixth Assessment Report of the Intergovernmental Panel on Climate Change (IPCC AR6). This dataset includes downscaled projections from ScenarioMIP model runs for which daily scenarios were produced and distributed through the Earth System Grid Federation. 
```

File: mock/datasets/cmip6-tas.data.mdx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    name: Mick Truyts
+    url: https://unsplash.com/photos/x6WQeNYJC1w
+taxonomy:
+  - name: Topics
+    values:
+      - Climate
+layers:
+  - id: combined_CMIP6_daily_GISS-E2-1-G_tas_kerchunk_TEST
+    stacCol: combined_CMIP6_daily_GISS-E2-1-G_tas_kerchunk_TEST
+    name: CMIP6 Daily GISS-E2-1-G Near-Surface Air Temperature
```
Comment (j08lue) [inline mock/datasets/cmip6-tas.data.mdx:19]:
```suggestion
    name: CMIP6 Daily GISS-E2-1-G Near-Surface Air Temperature (demo subset)
```

File: mock/datasets/cmip6-tas.data.mdx [hunk 5]
```diff
@@ -0,0 +1,76 @@
+---
+id: combined_CMIP6_daily_GISS-E2-1-G_tas_kerchunk_TEST
+name: 'CMIP6 Daily GISS-E2-1-G Near-Surface Air Temperature'
```
Comment (j08lue) [inline mock/datasets/cmip6-tas.data.mdx:3]:
```suggestion
name: 'CMIP6 Daily GISS-E2-1-G Near-Surface Air Temperature (demo subset)'
```

File: mock/datasets/cmip6-tas.data.mdx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  * Longitude Resolution: 0.25 degrees (25 km)
+  * Temporal Resolution: daily
+
+Source: [https://www.nccs.nasa.gov/services/data-collections/land-based-products/nex-gddp-cmip6](https://www.nccs.nasa.gov/services/data-collections/land-based-products/nex-gddp-cmip6)
+
+## AWS Public Dataset
+
+There are 2 datasets listed on the AWS Registry of Open Data for [NEX-GDDP-CMIP6](https://registry.opendata.aws/nex-gddp-cmip6/). First, there is an archive of NetCDF files from about 35 different climate models, each supplying historical and predicted values for up to 9 environment variables, daily, from 1950 to 2100. Second, there is an archive of COGs generated from the corresponding NetCDFs to support visualization via dynamic tiling using COGs. COGs are only available for 2 models. The COG archive also includes monthly aggregatations across those models.
+
+## About this dataset
```
Comment (j08lue) [inline mock/datasets/cmip6-tas.data.mdx:70]:
```suggestion
## This is a demo subset of NEX-GDDP-CMIP6
```

**Review by j08lue — COMMENTED** [review]
Looks great, @abarciauskas-bgse!

I made a few suggestions to edit the text to

1. clarify the use of the word "dataset" - both used for the original source ("NEX-GDDP-CMIP6") and for the VEDA copy of it
2. highlight more that this is just a demo subset - so users do not get surprised to find that it only covers 1950-1951.

Another question: I see you now point at the `dev` API - Is that necessary / do we not have the collection in staging? `dev` is probably a bit risky to use for anything user-facing, because we break it every now and then, right @slesaad?

**Comment by abarciauskas-bgse** [comment]
Thanks for looking @j08lue I think I have addressed your comments but I also think I could/should reach out to the dataset contacts Dr. Bridget Thrasher and Dr. Ian Brosnan, NASA NEX listed on https://www.nccs.nasa.gov/services/data-collections/land-based-products/nex-gddp-cmip6 to get their review and approval?

**Comment by slesaad** [comment]
@j08lue that's right. the next step, (also mentioned by aimee in the comment above) is to publish it to staging and create a PR in the veda-config repo. This is probably just an experimental PR.

File: mock/datasets/cmip6-tas.data.mdx (hunk 1 above)
Comment (abarciauskas-bgse) [reply mock/datasets/cmip6-tas.data.mdx]: I made a small change, since we are not hosting any of the data files themselves on VEDA

**Comment by abarciauskas-bgse** [comment]
@slesaad is correct, this PR was so that we could easily preview what the data would look like before I go about publishing it to the staging API.

So the next steps are:
- [ ] I noticed that the spatial extent values in [the dev collection](https://dev-stac.delta-backend.com/api/stac/collections/combined_CMIP6_daily_GISS-E2-1-G_tas_kerchunk_TEST) are not what I expect them to be so I'm going to address that 
- [ ] publish to staging STAC
- [ ] Add the STAC documentation to veda-docs and data processing documentation to veda-data
- [ ] open a PR in veda-config that looks almost exactly like this PR

**Comment by abarciauskas-bgse** [comment]
closing via https://github.com/NASA-IMPACT/veda-config/pull/323

**Pull request closed without merging**